File: algorithms/1_easy/T/03467_transform_array_by_parity/solution.cpp
class Solution {
public:
    vector<int> transformArray(vector<int>& nums) {

        int even_count{0};
        for (const int &num: nums) {
            if ((num & 1) == 0) {
                even_count += 1;
            }
        }

        const size_t len{nums.size()};
        for (int i{0}; i < len; ++i) {
            if (i < even_count) {
                nums[i] = 0;
            } else {
                nums[i] = 1;
            }
        }

        return nums;
    }
};

File: algorithms/1_easy/T/03467_transform_array_by_parity/solution.rs
impl Solution {
    pub fn transform_array(mut nums: Vec<i32>) -> Vec<i32> {
        let mut even_count = 0usize;
        for &num in & nums {
            if (num & 1) == 0 {
                even_count += 1;
            }
        }

        for i in 0..nums.len() {
            if i < even_count {
                nums[i] = 0;
            } else {
                nums[i] = 1;
            }
        }

        nums
    }
}
